src/Language/PureScript/Docs/Convert/Single.hs

```diff
@@ -159,6 +175,10 @@ convertDeclaration (P.ValueFixityDeclaration sa fixity (P.Qualified mn alias) _)
   Just . Right $ mkDeclaration sa title (AliasDeclaration fixity (P.Qualified mn (Right alias)))
 convertDeclaration (P.TypeFixityDeclaration sa fixity (P.Qualified mn alias) _) title =
   Just . Right $ mkDeclaration sa title (AliasDeclaration fixity (P.Qualified mn (Left alias)))
+convertDeclaration (P.KindDeclaration sa kindSig _ tys) title =
+  Just $ Left ([(title, AugmentType), (title, AugmentClass)], AugmentKindSig comms kindSig (tys $> ()))
```

Comment:
Using `AugmentType` here means `ExternDeclaration`s are also considered when matching an augment with its corresponding declaration. Will that cause a problem? Or should I add a new AugmentType specifically for this concern?

Reply:
I'm not sure. Why might it cause a problem?

Reply:
In case `isType` is ever updated in the future to include other potential declarations. Since this should only match on 4 declarations (data, type, newtype, class), I was wondering whether it might be better to create a separate function (e.g. `isKindAugmentable`) specifically for those and indicate that via a new `AugmentType` (e.g. `AugmentWithKind`) that uses that `isKindAugmentable` function.

Reply:
Here's a link to [`isType`](https://github.com/purescript/purescript/pull/4100/files#diff-03c8dee30986517880ff71bf3ac056c8caec48c7bebef10fa1e7fb30c6b01d7fR257)

Reply:
I'm not sure I completely understand. Are you imagining that if there's an ExternDeclaration and there is somehow also a kind signature for a declaration with the same name elsewhere in the module, that the kind signature would be incorrectly attached to the ExternDeclaration?

Reply:
I'd be inclined to say that it shouldn't matter because the name of an ExternDeclaration always starts with a lowercase letter, whereas the names of anything you can give kind signatures to (data types, classes) always start with uppercase letters, and so they can't ever overlap.

Reply:
I think my question is probably a rabbit trail and not work thinking about further. However, let me try to explain my flow of thinking.

Kind signature declarations should only be matched to a data type / newtype (i.e. DataDeclaration), type synonym (i.e. TypeSynonymDeclaration), or a type class (i.e. TypeClassDeclaration) declaration. Kind signatures are never associated with FFI declarations (e.g. `ExternDataDeclaration`).

The `AugmentType` values (e.g. `AugmentClass` and `AugmentType`) merely [indicate which function to use](https://github.com/purescript/purescript/blob/c7cf026479788c74ed8190c6f6c4d97711217001/src/Language/PureScript/Docs/Convert/Single.hs#L95-L96) to determine if an augmentation matches a given declaration. 

Since we're currently using [`AugmentType` for the `KindDeclaration`](https://github.com/purescript/purescript/blob/c7cf026479788c74ed8190c6f6c4d97711217001/src/Language/PureScript/Docs/Convert/Single.hs#L185), the resulting function, [`isType`, will also return true when given a `ExternDataDeclaration`](https://github.com/purescript/purescript/blob/c7cf026479788c74ed8190c6f6c4d97711217001/src/Language/PureScript/Docs/Types.hs#L257). Yes, this isn't a problem now because `ExternDeclaration` is always lowercase whereas the other three declaration types start with uppercase letters. However, if we make changes later (e.g. casing changes; a new declaration is added that also happens to be lowercase) and somehow this declaration should happen to slip through, wouldn't this cause a bug?

So, my question is should we update the types to show something like this:
```haskell
-- 1. Define a new type that will
-- use a different matching function for 
-- augmenting declarations with kind signatures
-- | Different declarations we can augment
data AugmentType
  = AugmentClass
  -- ^ Augment documentation for a type class
  | AugmentType
  -- ^ Augment documentation for a type constructor
  | AugmentWithKindSignature
  -- ^ Augment data/newtype/type/class declaration with a kind signature

-- 2. Update the `matches` function in the `augmentDeclarations` code
-- to use this new function called `isKindAugmentable`
matches d (name, AugmentType) = isType d && declTitle d == name
matches d (name, AugmentClass) = isTypeClass d && declTitle d == name
matches d (name, AugmentWithKindSignature) = 
   isKindAugmentable d && declTitle d == name
   
-- 3. In Docs/Types.hs, add a new function 
-- that only returns True on those declarations
-- that can be augmented by a kind signature
isKindAugmentable :: Declaration ->  Bool
isKindAugmentable Declaration{..} =
  case declInfo of
    TypeSynonymDeclaration{} -> True
    DataDeclaration{} -> True
    TypeClassDeclaration{} -> True
    -- notice how ExternDataDeclaration returns False here
    _ -> False
```
```diff
 -- 4. Update the convertDeclaration 
 -- for kind signatures to use this new
 -- function
convertDeclaration (P.KindDeclaration sa kindSig _ tys) title =
- Just $ Left ([(title, AugmentType), (title, AugmentClass)], AugmentKindSig ksi)
+ Just $ Left ([(title, AugmentWithKindSignature)          ], AugmentKindSig ksi)

Reply:
Or am I just overthinking this?

Reply:
I think this would be unnecessarily cautious. Changing the syntax so that types can start with lowercase letters is almost certainly never going to happen, and the compiler should reject attempts to give `foreign import data` declarations separate kind signatures anyway (because the `foreign import data` declaration itself already contains a kind signature).

Reply:
Ok. Thanks for clarifying!